.github/workflows/landblocking_tcb.yml

```diff
@@ -0,0 +1,244 @@
+name: clippy
+
+# Why `on: repository_dispatch`?
+#
+# If we trigger this action on pull_request events,
+# then it will run with the forked repo GITHUB_TOKEN
+# this is a problem as it won't allow the action to write
+# a comment on the PR in the last step.
+#
+# instead, we trigger this remotely via a bot we're running ourselves.
+on:
+  repository_dispatch:
+    types: [new_PR]
+
+jobs:
+  run_clippy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Who triggered this?
+        env:
+          PULL_ID: ${{ github.event.client_payload.pull_id }}
+        run: |
+          echo "action triggerd by https://github.com/libra/libra/pull/$PULL_ID"
+
+      - name: Whitelist of users
+        uses: actions/github-script@0.5.0
+        env:
+          PULL_ID: ${{ github.event.client_payload.pull_id }}
+        with:
+          github-token: ${{secrets.MIRAI_BOT}}
+          script: |
+            const fs = require('fs');
+            var path = require('path');
+
+            async function main() {
+              // get info on PR
+              const pullrequest = await github.pulls.get({
+                owner: "libra",
+                repo: "libra",
+                pull_number: process.env.PULL_ID
+              });
+
+              // allowlist
+              var allowlist = ["mimoo", "anomalroil", "xvschneider"];
+              if (!allowlist.includes(pullrequest.data.user.login)) {
+                console.log(pullrequest.data.user.login + " is not in the allowlist of this github action");
+                throw "user not in whitelist!";
+              }
+            }
+
+            main();
+
+      - name: Checkout the PR
+        if: success()
+        uses: actions/checkout@v2
+        with:
+          path: "libra"
+          fetch-depth: 50 # this is to make sure we obtain the target base commit
+          repository: ${{ github.event.client_payload.owner }}/${{ github.event.client_payload.repo }} # repo of the forked libra/libra
+          ref: ${{ github.event.client_payload.pull_ref }} # commit that triggered the PR
+
+      - name: Get files that were changed
+        if: success()
+        working-directory: ./libra
+        env:
+          BASE: ${{ github.event.client_payload.base_ref }}
+        run: |
+          git diff -U0 $BASE | tee ../files_changed
+
+      - name: run Clippy
+        if: success()
+        working-directory: ./libra
+        timeout-minutes: 60
+        run: |
+          set -x
+
+          # update rust and get clippy
+          rustup install nightly
+          rustup update
+          rustup component add clippy --toolchain nightly
+
+          # run clippy "allow" lints in "warn" mode
+          # last lint update: https://github.com/rust-lang/rust-clippy/blob/master/CHANGELOG.md#rust-148
+          cargo +nightly clippy --quiet --message-format=json --workspace -- \
```

Comment:
@mimoo  I think I can add here the target directories (that are part of TCB). which were those again? consensus, libra_node, secure ?